components/src/core/Chip/Chip.tsx

```diff
@@ -0,0 +1,192 @@
+ * @prop {string} [textColor] - The color of the text content in the Chip.
+ * @prop {string} [chipColor] - The background color of the Chip.
+ * @prop {string} [borderColor] - The border color of the Chip.
+ * @prop {boolean} [selected] - Whether the Chip is in a selected state.
+ * @prop {boolean} [disabled] - Whether the Chip is in a disabled state.
+ * @prop {$DeepPartial<MD3Theme>} [theme] - Theme value overrides specific to this component.
+ * @prop {React.ReactElement} [avatar] - Avatar component to be displayed in the Chip.
+ * @prop {string} [mode='outlined'] - Chip mode, either 'outlined' or 'elevated'.
+ */
+
+/**
+ * A customizable Chip component.
+ *
+ * @param {ChipProps} props - The props for the Chip component.
+ * @returns {JSX.Element} - The rendered Chip component.
+ */
+
+export type ChipProps = Omit<PaperChipProps, 'icon' | 'mode' | 'selectedColor'> & {
+    theme?: $DeepPartial<MD3Theme>;
+    iconColor?: string;
+    textColor?: string;
+    chipColor?: string;
+    borderColor?: string;
+    icon?: IconSource;
+    mode?: 'elevated' | 'outlined'; // Updated modes
+    avatar?: React.ReactElement; // New prop for passing Avatar component
+};
+
+export const Chip: React.FC<ChipProps> = (props) => {
+    const {
+        children,
+        icon,
+        iconColor,
+        style,
+        textStyle,
+        mode = 'outlined',
+        selected,
+        disabled,
+        theme: themeOverride,
+        avatar,
+        chipColor,
+        borderColor,
+        textColor,
+        ...rest
+    } = props;
+
+    //@ts-ignore
+    const theme = useTheme(themeOverride) as MD3Theme;
+
+    const isOutlined = mode === 'outlined';
+    const isElevated = mode === 'elevated';
+
+    // @TODO update the disabled colors once the condition is set in place
+    const defaultChipColor = isOutlined
+        ? disabled
+        //the chip background color should be transparent in case it is set to disable in outline mode
+            ? 'transparent'
+            : selected
+            //the chip background color should be primary[80] in case it is set to selected in outline mode
+            ? theme.colors.primaryContainer
+            //the chip background color should be transparent in case it is set to unselected in outline mode
+            : 'transparent'
+        : isElevated
+        ? disabled
+            ?
+            //the chip background color should be neutral[10] 5% in case it is set to disable in elevated mode 
+            //@ts-ignore
+              theme.colors.disabledContainer
+            : selected
+
+            //the chip background color should be primary[80] in case it is set to selected in elevated mode
+            ? theme.colors.primaryContainer
+
+            //the chip background color should be neutral[97] in case it is set to unselected in elevated mode
+            : //@ts-ignore
+              theme.colors.surfaceContainerLow
+        : undefined;
```

Comment:
Why do we have this line, when the `mode` can either be `outlined` or `elevated`?

Reply:
updated the code